styles/style_ptr.h

```diff
@@ -3,6 +3,17 @@
 
 #include "blade_style.h"
 
+template<class BASE, class L1> class Compose;  // forward-declare Layers node
+
+namespace style_base_check_detail {
+  template<class T> struct TopBase { using type = T; };
+  template<class B, class L> struct TopBase<Compose<B,L>> : TopBase<B> {};
```

Comment:
What's the point of checking the base layer rather than the whole stack? The result should be the same.

Reply:
We only care about the first layer, right? (for this error since that's what causes a compile fail?)
It must be solid (or Int<0>). Later solids don’t fix a transparent base, so the check enforces that rule.

Reply:
What we care about is that the result is solid.
Digging into why it's solid seems like extra work.

Reply:
The result-only check hides the error from color_details. It only inspects the final composite type, so if the base is transparent but a later solid makes the result opaque, IsOpaqueColor says “ok” and our “must be solid” message never fires.

That’s why I kept the base check: it flags the illegal transparent base immediately, at the point of mistake.

Reply:
"a later solid makes the result opaque"

I'm suggesting that we make this a separate compilation error (in layers.h)